keras_preprocessing/image/directory_iterator.py

```diff
@@ -147,3 +147,8 @@ def filepaths(self):
     @property
     def labels(self):
         return self.classes
+
+    @property  # mixin needs this property to work
+    def sample_weight(self):
+        # no sample weights will be returned
+        return None
```

Comment:
Suggestion: Wouldn't returning an array of ones be cleaner?

Reply:
you might be right. Let me take a look how `NumpyArrayIterator` does it, but I think it does something similar, as in not returning any weights at all, hence saving some computation when calculating the loss function down the line.